hphp/runtime/vm/jit/relocation-arm.cpp

```diff
@@ -497,120 +509,108 @@ size_t relocateImpl(RelocationInfo& rel,
+    for (auto src = Instruction::Cast(env.start);
+         src < Instruction::Cast(env.end);
+         src += srcCount * kInstructionSize) {
+      auto const srcAddr = reinterpret_cast<TCA>(src);
+      auto destAddr = env.destBlock.frontier();
+      auto const dest = Instruction::Cast(destAddr);
+      srcCount = 1;
+      destCount = 1;
 
       // Initially copy the instruction word
-      auto dest = dest_block.frontier();
-      dest_block.bytes(kInstructionSize, src);
-      size_t src_count = 1, dest_count = 1;
+      env.destBlock.bytes(kInstructionSize, srcAddr);
 
       // If it's not a literal, then attempt any special relocations
-      if (!literals.count(Instruction::Cast(src)) &&
-          !relocateSmashableHelper(rel, meta, dest_block,
-                                   start, end, src, dest, jmp_dest,
-                                   src_count, dest_count, far, rewrites,
-                                   internal_refs_need_update) &&
-          !relocatePCRelativeHelper(dest_block,
-                                    start, end, src, dest, jmp_dest,
-                                    src_count, dest_count, far, rewrites,
-                                    internal_refs_need_update) &&
-          !relocateImmediateHelper(rel, dest_block, end,
-                                   src, dest,
-                                   src_count, dest_count, rewrites,
-                                   internal_refs_need_update)) {
+      if (!literals.count(src) &&
+          !relocateSmashable(env, src, dest, srcCount, destCount) &&
+          !relocatePCRelative(env, src, dest, srcCount, destCount) &&
+          !relocateImmediate(env, src, dest, srcCount, destCount)) {
         // Do nothing, as the instruction word was initially copied above
       }
 
-      if (src == start) {
+      if (srcAddr == env.start) {
         /*
          * For the start of the range, we only want to overwrite the "after"
          * address (since the "before" address could belong to the previous
          * tracelet, which could be being relocated to a completely different
          * address. recordRange will do that for us, so just make sure we
          * have the right address setup.
          */
-        dest_start = dest;
+        destStart = destAddr;
       } else {
-        rel.recordAddress(src, dest, 0);
+        env.rel.recordAddress(srcAddr, destAddr, 0);
       }
 
-      src += src_count * kInstructionSize;
-      dest += dest_count * kInstructionSize;
-      asm_count += dest_count;
-      assertx(dest <= dest_block.frontier());
-      dest_block.setFrontier(dest);
-    } // while (src != end)
+      // Update the destAddr and reset the frontier
+      destAddr += destCount * kInstructionSize;
+      assertx(destAddr <= env.destBlock.frontier());
+      env.destBlock.setFrontier(destAddr);
 
-    if (exit_addr) {
-      *exit_addr = jmp_dest;
-    }
+      asmCount += destCount;
+    } // while (src != env.end)
```

Comment:
This comment is a lie now, and also most editors make it pretty easy to chase the start of a scope.